### PR TITLE
Bugfix: trello state bugfix

### PR DESCRIPTION
**Pull request by lfbdev**
# Summary

## Description

1.  state file resource option: workflow dir lacks "/"
2. The board must be found in trello precisely, so we must get the boardid from state resource.
- if board found, list id(todo/doing/done) can be queried
3. add "kanban" in doc, so that to let user config the kanban name of trello

### CODE REVIEW

**Review by daniel-hutao — CHANGES_REQUESTED** [review]

File: examples/config.yaml [hunk 1]
```diff
@@ -45,6 +45,8 @@ tools:
     repo: golang-demo
     api:
       name: trello
+      category: trello-github-integ
+      ident: trello-github-integ-golang-demo
```
Comment (IronCore864) [inline examples/config.yaml:49]: What does ident mean here? Ident normally means identification; does it mean identification here? Is the word used by Trello official doc so we use the abbreviation？More comments in the readme of this plugin would help.

File: pkg/util/trello/trello.go [hunk 2]
```diff
@@ -44,27 +50,58 @@ func (c *Client) CreateList(board *trello.Board, listName string) (*trello.List,
...
 }
+
+func genBoardId(ident, category string) (string, error) {
+	// use default local backend for now.
+	b, err := backend.GetBackend(backend.BackendLocal)
+	if err != nil {
+		return "", err
+	}
+	// create a state manager using the default local backend
+	smgr, err := statemanager.NewManager(b)
```
Comment (IronCore864) [inline pkg/util/trello/trello.go:91]: Plugins should be state-agnostic.

**Review by IronCore864 — CHANGES_REQUESTED** [review]

File: docs/plugins/trello-github-integ_plugin.md [hunk 3]
```diff
@@ -11,6 +11,9 @@ _This plugin depends on the following two environment variables:_
 
 Set the values accordingly before using this plugin.
 
+## 3 Tips:
+_Trello board description managed by DevStream, don't modify it_
```
Comment (IronCore864) [inline docs/plugins/trello-github-integ_plugin.md:15]: Trello board description “is” managed by DevStream, please don't modify it.

Missing "is"; add "please" to be more polite; and missing period at the end.

File: pkg/util/trello/trello.go [hunk 4]
```diff
@@ -44,27 +50,41 @@ func (c *Client) CreateList(board *trello.Board, listName string) (*trello.List,
...
-		lists, err := b.GetLists()
-		if err != nil {
-			return nil, err
+		if checkTargetBoard(owner, repo, kanban, b) {
+			lists, err := b.GetLists()
+			if err != nil {
+				return nil, err
+			}
+			if len(lists) != DefaultListsNumber {
+				log.Errorf("Unknown lists format: len==%d", len(lists))
```
Comment (IronCore864) [inline pkg/util/trello/trello.go:69]: Missing period at the end of the log.

File: pkg/util/trello/trello.go [hunk 5]
```diff
@@ -44,27 +50,41 @@ func (c *Client) CreateList(board *trello.Board, listName string) (*trello.List,
...
+
+func checkTargetBoard(owner, repo, kanban string, b *trello.Board) bool {
+	if b.Name == kanban && b.Desc == boardDesc(owner, repo) {
+		return true
+	}
+	return false
+}
+
+func boardDesc(owner, repo string) string {
+	return fmt.Sprintf("Description managed by DevStream,don't change.%s/%s", owner, repo)
```
Comment (IronCore864) [inline pkg/util/trello/trello.go:89]: "Description is managed by DevStream, please don't modify. %s/%s"

or simply

"Managed by DevStream, please don't modify. %s/%s"

- There should be a space after punctuation.
- change -> modify (to be consistent with the readme)
- add "please"

**Review by IronCore864 — APPROVED** [review]

**Review by daniel-hutao — APPROVED** [review]

**Pull request merged**